packages/excalidraw/actions/actionFinalize.tsx

```diff
@@ -51,6 +51,7 @@ export const actionFinalize = register({
             cursorButton: "up",
             editingLinearElement: null,
             selectedLinearElement: null,
+            selectionElement: null,
```

Comment:
@s1lverfish what is the purpose of this change?

Reply:
Nevermind, put it together.

Reply:
In `onPointerUpFromPointerDownHandler` the `selectionElement` is set to `null`. The problem is that in actionFinalize this update get's lost.
```
  private onPointerUpFromPointerDownHandler(
    pointerDownState: PointerDownState,
  ): (event: PointerEvent) => void {
    return withBatchedUpdates((childEvent: PointerEvent) => {
      
...

      this.setState((prevState) => ({
        isResizing: false,
        isRotating: false,
        resizingElement: null,
        selectionElement: null, //<---------- selectionElement is set to null here

...       

      if (this.state.editingLinearElement) {
        if (
          !pointerDownState.boxSelection.hasOccurred &&
          pointerDownState.hit?.element?.id !==
            this.state.editingLinearElement.elementId
        ) {
          this.dismissLinearEditor(); //<---------- actionFinalize is being called here
        } else {
```

Here is a version where I don't set the `selectionElement` to `null` in `actionFinalize`:

https://github.com/user-attachments/assets/cbce829a-c99c-4d9b-8a1a-d6e8ec425393

If I log out the `appState` there is a `selecitonElement` that has width and height of 0.
I'm not exactly sure how this causes the main problem I ran into (not being able to click on anything), but I think either way it's not good to have a 0 by 0 `selectionElement` floating around in the `appState`.

Reply:
Yeah, I put it together eventually :)